src/Beberlei/Metrics/Collector/Null.php

```diff
@@ -13,6 +13,9 @@
 
 namespace Beberlei\Metrics\Collector;
 
+/**
+ * @deprecated for PHP7 compatibility, use NullCollector instead.
+ */
 class Null implements Collector
```

Comment:
ok for the deprecation.

About the `extends`: Less code == less bug; Then is easier to understand, to maintain.